### PR TITLE
to buy one product

### PR DESCRIPTION
**Pull request by olenashevchuk**

### CODE REVIEW

File: src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js [hunk 1]
```diff
@@ -80,17 +87,54 @@ const ProductAdvancedView = (props) => {
...
+        balance: wallet.balance - product.price * product.quantity
+      })
       setStatusMessage({
         open: true,
         message: `Product was successfully moved to ${actionCollection}`,
         type: 'success'
       })
+      /*
+        delete current product from collection card */
+      await deleteData(COLLECTIONS.CART, id).then(() => history.goBack())
```
Comment (GingerBoy14) [inline src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js:133]: Use or `await`, or `then`(await - better, just move history push from then under await call)

File: src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js [hunk 2]
```diff
@@ -99,15 +143,21 @@ const ProductAdvancedView = (props) => {
...
-        </DropdownItem>
+        <WalletCombinedWithSelect
+          setStatusMessage={setStatusMessage}
+          onSubmitFunction={handleMoveProduct}
+          onClick={onCheckClick}>
+          <DropdownItem divider>
+            <Typography>{firstElement}</Typography>
+          </DropdownItem>
+        </WalletCombinedWithSelect>
       )}
```
Comment (GingerBoy14) [inline src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js:161]: When u move product from wish to cart u'll also need to select the wallet?

File: src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js [hunk 3]
```diff
@@ -80,17 +87,54 @@ const ProductAdvancedView = (props) => {
...
+  const onCheckClick = () => {
+    let status = true
+    //required fields
+    const fields = ['name', 'price', 'quantity']
+    for (let field of fields) {
+      data.hasOwnProperty(field)
+        ? data[field]
+          ? (status = status && true)
+          : (status = false)
+        : (status = false)
```
Comment (GingerBoy14) [inline src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js:100]: Can u try this one: status = `data.hasOwnProperty(field) && data[field] && status ? true : false`?

**Review by GingerBoy14 — CHANGES_REQUESTED** [review]

File: src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js [hunk 4]
```diff
@@ -225,8 +300,8 @@ ProductAdvancedView.propTypes = {
   purchasedDate: PropTypes.number,
   categoryBalance: PropTypes.number,
   reminderDate: PropTypes.number,
-  assignedUser: PropTypes.string,
-  type: PropTypes.oneOf(Object.keys(productTypeMap)).isRequired
+  assignedUser: PropTypes.string
+  //type: PropTypes.oneOf(Object.keys(productTypeMap)).isRequired
```
Comment (GingerBoy14) [inline src/app/domains/Product/components/views/ProductAdvancedView/ProductAdvancedView.template.js:304]: If u don't use it - remove

**Review by GingerBoy14 — APPROVED** [review]

**Review by okrdima — APPROVED** [review]

**Pull request merged**